packages/flutter/test/material/checkbox_test.dart

```diff
@@ -603,6 +604,56 @@ void main() {
     expect(box.size, equals(const Size(60, 36)));
   });
 
+  testWidgets('Checkbox stops hover animation when removed from the tree.', (WidgetTester tester) async {
+    const Key checkboxKey = Key('checkbox');
+    bool checkboxVal = true;
+
+    await tester.pumpWidget(
+      Theme(
+        data: ThemeData(materialTapTargetSize: MaterialTapTargetSize.padded),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: Material(
+            child: Center(
+              child: StatefulBuilder(
+                builder: (_, StateSetter setState) => Checkbox(
+                  key: checkboxKey,
+                  value: checkboxVal,
+                  onChanged: (bool newValue) => setState(() {checkboxVal = newValue;}),
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(find.byKey(checkboxKey), findsOneWidget);
+    final Offset checkboxCenter = tester.getCenter(find.byKey(checkboxKey));
+    final TestGesture testGesture = await tester.createGesture(kind: PointerDeviceKind.mouse);
+    await testGesture.moveTo(checkboxCenter);
+
+    await tester.pump(); // start animation
+    await tester.pump(const Duration(milliseconds: 25)); // hover animation duration is 50 ms. It is half-way.
+
+    await tester.pumpWidget(
+      Theme(
+        data: ThemeData(materialTapTargetSize: MaterialTapTargetSize.padded),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: Material(
+            child: Center(
+              child: Container(),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await testGesture.removePointer();
```

Comment:
Add 
```dart
      // Hover animation should not trigger an exception when the checkbox is removed
      // before the hover animation should complete.
      expect(tester.takeException(), isNull);
```

This signifies to future readers or contributors that the point of this test is that an exception does not occur in the above situation.

Reply:
@shihaohong added.